packages/smooth_app/lib/pages/scan/mkit_scan_helper.dart

```diff
@@ -233,13 +233,23 @@ class _MLKitScanDecoderIsolate {
 
     port.send(
       barcodes
-          .map((Barcode barcode) => barcode.rawValue)
+          .map((Barcode barcode) => _changeBarcodeType(barcode))
           .where((String? barcode) => barcode?.isNotEmpty == true)
           .cast<String>()
           .toList(growable: false),
     );
   }
 
+  static String? _changeBarcodeType(Barcode barcode) {
+    //EAN13 begins with 0 is detected as UPC-A by google_ml_barcode_scanner v0.0.2
+    if (barcode.format == BarcodeFormat.upca) {
+      if (barcode.rawValue != null) {
```

Comment:
You should merge the two if into a single one

Reply:
> You should merge the two if into a single one

✅